### ISSUE TITLE
Does it work with the latest VS .NET 6

### ISSUE THREAD

**Issue by fasteddys** [jdanylko/Menu-System-w-Identity#1]
Hi I am trying out your projects, they all seem very relevant and basic stuff, wonder why others dont have this main stream samples.

Can you share how to run this in latest .NET 6, also what is the difference between MVC and Razor and Blazor pages, will it work with all of them.

**Comment by jdanylko** [comment]
I'm in the process of digging more into .NET 6 and what it means to my past, current, and future projects.

Rest assured...I have at least four projects utilizing this technique and will be updating this technique in a future post. Since I did update it to ASP.NET Core 3.1 (found [here](https://www.danylkoweb.com/Blog/updating-the-menu-system-to-aspnet-core-31-with-microsoft-identity-N7)), I'm hoping the .NET 6.0 shouldn't be too big of a change.

As for the difference between MVC, Razor, and Blazor pages, this technique focuses on server-side technologies (like Microsoft Identity and EF), but it still returns menu item data to the client whether it's an MVC application or Blazor application.

The benefit of this is the ability to use any one of these client-side technologies to implement a menu system based on what's returned from the controller (or lately...[Razor Pages](https://docs.microsoft.com/en-us/archive/msdn-magazine/2017/september/asp-net-core-simpler-asp-net-mvc-apps-with-razor-pages)). I just decided to use ViewComponents to encapsulate the menu data into one component for re-use.

With Blazor, it uses WebAssembly under the hood to make fast client frontsends with .NET/C#. This would simply use the menu options of what they can see (and not see) returned to create a client-side menu system in Blazor.

**Bottom line:** the method above can be used for any type of client-side system. It can even be implemented as an API of valid menu items for a user.

(Sorry...got carried away) :-)

**Comment by fasteddys** [comment]
that's great design by making the menu itself a view component, I read up and your points are wise, 3.1 is not very different I am guessing.

Pingback to you, I just discovered your website, you cover such interesting topics, love it!
